### PR TITLE
feat: experimental indent when logging to terminal

### PR DESCRIPTION
**Pull request by wolfv**
TODO:

- [x] refactor `run_in_env` so that we can use it both in tests and build execution
- [x] Add more `span!` (e.g. when running the tests)
- [x] make interpreter selectable while we are at it :)
- [x] Test `rebuild` and `test` subcommands
- [x] Add "summary" function after the build(s) is done

Still needs fixing:

- [x] Summary is nice, but needs some more tweaks:
  - Add a Github subscriber that prints warning as `::warning::{message}` and error as `::error::{message}`
  - Make the summary have proper tables

### CODE REVIEW

**Comment by wolfv** [comment]
It's starting to look cool: https://asciinema.org/a/640577

File: src/render/solver.rs [hunk 1]
```diff
@@ -134,7 +133,7 @@ pub async fn create_environment(
                     platform,
                     &repodata_cache,
                     download_client.clone(),
-                    tool_configuration.multi_progress_indicator.clone(),
+                    tool_configuration.fancy_log_handler.clone(),
```
Comment (schuylermartin45) [inline src/render/solver.rs:136]: Amazing name lol

**Review by schuylermartin45 — APPROVED** [review]
Pretty straight forward, looks good.

**Comment by pavelzw** [comment]
The "install yq with chocolatey" ci step seems to fail quite often... We could do `prefix-dev/setup-pixi` and afterwards `pixi global install yq`...

**Pull request merged**